content/1_General/Cpp_Command.mdx

```diff
@@ -286,32 +302,32 @@ Of course, retyping the flags above can get tedious. You should define shortcuts
 	></Resource>
 </Resources>
 
-First, create your `.zshenv` if it doesn't already exist.
+First, create your `.zshrc` if it doesn't already exist.
 
 ```
-touch ~/.zshenv
+touch ~/.zshrc
 ```
 
-Open your `.zshenv` with TextEdit.
+Open your `.zshrc` with a text editor.
 
 ```
-open -a TextEdit ~/.zshenv
+open ~/.zshrc
 ```
 
 or some text editor (ex. sublime text with `subl`).
 
 ```
-subl ~/.zshenv
+subl ~/.zshrc
 ```
 
 You can add **aliases** and **functions** here, such as the following to compile and run C++ on Mac.
 
 ```
-co() { g++ -std=c++17 -O2 -o $1 $1.cpp -Wall; }
-run() { co $1 && ./$1 & fg; }
+co() { g++ -std=c++17 -O2 -o "${1%.*}" $1 -Wall; }
+run() { co $1 && ./${1%.*} & fg; }
```

Comment:
Isn't the original shell script preferable since it safes you a few keystrokes (you don't have to type .cpp)?

Or is there something I'm missing? (oops I don't use shell scripts)

Reply:
As hinted at in the commit message, this makes completion easier (tab completion normally works by files). The original method would imply completing and then deleting the `.cpp` from the end if that makes sense.